structuretoolkit/analyse/symmetry.py

```diff
@@ -345,25 +345,36 @@ def get_primitive_cell(
         >>> len(symmetry.get_primitive_cell()) == len(basis)
         True
         """
+        if not all(self._structure.pbc):
+            raise ValueError("Can only symmetrize periodic structures.")
         ret = spglib.standardize_cell(
             self._get_spglib_cell(use_elements=use_elements, use_magmoms=use_magmoms),
             to_primitive=not standardize,
         )
         if ret is None:
             raise SymmetryError(spglib.spglib.spglib_error.message)
-        cell, positions, indices = ret
-        positions = (cell.T @ positions.T).T
-        new_structure = self._structure.copy()
-        new_structure.cell = cell
-        new_structure = new_structure[: len(indices)]
+        cell, scaled_positions, indices = ret
         indices_dict = {
             v: k
             for k, v in structuretoolkit.common.helper.get_species_indices_dict(
                 structure=self._structure
             ).items()
         }
-        new_structure.symbols = [indices_dict[i] for i in indices]
-        new_structure.positions = positions
+        symbols = [indices_dict[i] for i in indices]
+        arrays = {
+            k: self._structure.arrays[k]
+            for k in self._structure.arrays
+            if k not in ("numbers", "positions")
+        }
+        new_structure = type(self._structure)(
+            symbols=symbols,
+            scaled_positions=scaled_positions,
+            cell=cell,
+            pbc=[True, True, True],
+        )
+        for k, a in arrays.items():
+            new_structure.arrays[k] = a
```

Comment:
I don't really understand how this could be working. The new structure has a different number of atoms, right? Why shouldn't `array` be shortened accordingly?

Reply:
Correct I didn't think about that, but now that I do, I'm not sure if we can even do that in general.  I'm inclined to just drop user defined arrays during this and emit a warning.  What do you think?

Reply:
ok I can live with that